### ISSUE TITLE
VIRA-213: Make the vira_report and vira_menu readonly

### ISSUE THREAD

**Issue by chinwobble** [n0v1c3/vira#47]
The vira_report and vira_menu are readonly.

Based on my usage, you have created vira_prompt so that you can edit each section of a jira individually.

Like in fugitive in the git status screen you aren't allowed to edit text.

**Comment by n0v1c3** [comment]
We do have issues to edit them in place but for now you can use the `<cr>` to edit and select fields.

The report is the place for editing and menus for selecting. I do have an issue for editing and bulk editing in the menus.

**Comment by n0v1c3** [comment]
Async updates will now be applied to menus and reports bit this may not be 100% easy.

**Comment by n0v1c3** [comment]
Once they update async next step is to send info without and extra window then have the fake read only work. I am sure there is a way to toggle the read only once I actually look for it.

**Comment by n0v1c3** [comment]
A note for myself:
```
set nomodifiable
set modifiable
```

**Comment by n0v1c3** [comment]
@chinwobble I have done a merge one my VIRA-69 for this issue NOT the proper way, right away onto dev and risk onto master and that also made me remember this was not too much fun. There will be a proper way to do it especially now that I can multitask in vim and nvim. I will just need to turn it on and off while hide it from the display. This will also solve another one of your issues to be able to hide to the report titles. I am sure this will become upgrade them into these report titles but that will be the next great step after this.

**Comment by n0v1c3** [comment]
@chinwobble VIRA-213 should now how a readonly as needed. I shill needed to make it pretty but that can also be merged into the hiding of the entire "tab's" ugliness.

Let me know how it goes for you.

**Comment by chinwobble** [comment]
should this be nomodifable instead?
https://github.com/n0v1c3/vira/commit/5c1a3aad80c7f289807daea4e13a54bbf083b686#diff-93336de9847f9c412020ad62cdbfe420c7f41ce036616722624750f64b44c28eR48

**Comment by n0v1c3** [comment]
Yup, I just need to hide the message.

**Comment by n0v1c3** [comment]
I added one line of code to make make it a bit better but something is wrong deeper down. It is no longer styling the REPORT the way it was.

**Comment by n0v1c3** [comment]
@chinwobble that VIRA-213 push should definitely pass for this now.

**Comment by chinwobble** [comment]
The UI is very slow for me now.

Here's how you can profile the python code.

```viml
python3 << EOF
import vim
import cProfile
with cProfile.Profile() as pr:
  Vira.api.connect(vim.eval('g:vira_serv'))
pr.dump_stats('/tmp/vira_connect_stats.log')
EOF
```

Note you must not have any white space before / indents.

**Comment by n0v1c3** [comment]
This one is working for me. Only difference I will make is move it into `python` but you can do a quick push into it so I don't forget.

**Comment by chinwobble** [comment]
see also
https://thoughtbot.com/blog/profiling-vim

**Comment by n0v1c3** [comment]
I will take a closer look at it for sure.

**Comment by n0v1c3** [comment]
@chinwobble I believe the issue here is closed and only remains with the async issue.

**Comment by n0v1c3** [comment]
What I did proce more than it fixed so I removed a line and need to reopen this issue.

**Comment by n0v1c3** [comment]
Updated the issued number to break it free from the general open the doors and `README` issue.

**Comment by n0v1c3** [comment]
I did a push that covers the start if this one again and the small stuff starts to appear on me. VIRA-282 branch exists for now. I am sure more will come but here is a list for now:

```
- [ ] Deleted `ONE` line when the OTHER window was opened.
    - Open `report` back to `menu` delete a row.
- [ ] Reload the same Report with no changes.
- [ ] Reload the same `window` is missing.
    - `menu` or `report` does this to you
    - Any menu regardless of different data.
```

**Comment by n0v1c3** [comment]
Mostly into true code cleanup and hunt for bugs then it should be all good.

**Comment by n0v1c3** [comment]
@chinwobble this is on the `dev` branch for a test now if you want to give it a run. I am bug hunting and checking my other small issues before I will just get it to `master`.

**Comment by n0v1c3** [comment]
- [ ] `issues` fail to refresh if open when commenting.
    - This is the major one now.
- [ ] Keep `win` scroll position.
- [ ] Same `pos` for comment updates.
- [ ] Switch menus without close keeps position
    - Each menu should hold position
    - In fact each report should be in a history and hold its position in history.

**Comment by n0v1c3** [comment]
@chinwobble,

Check out the `dev` branch. I am sure I have gone to far with it but it should also be more than the original issue of `readonly`.

I will close this one and on to the other ones as long as it is working for some other people.
